funflow/src/Funflow/Tasks/Docker.hs

```diff
@@ -69,10 +67,13 @@ data DockerTaskInput = DockerTaskInput
 
 instance Semigroup DockerTaskInput where
   DockerTaskInput{ inputBindings = vols1, argsVals = args1 } <> DockerTaskInput{ inputBindings = vols2, argsVals = args2} = 
-    let agg (seen, xs) x = if Set.member x seen then (seen, xs) else (Set.insert x seen, x:xs)
+    -- TODO: deduplicate? If so, this becomes lawless unless, e.g., the collection of bindings is sth like Set.
+    {-let agg (seen, xs) x = if Set.member x seen then (seen, xs) else (Set.insert x seen, x:xs)
```

Comment:
You know, this is actually a good point I hadn't thought about when we were discussing this earlier. 

> docker run -v $PWD/dummy:/mnt1 -v $PWD/dummy:/mnt2 -v $PWD/test:/mnt1  python:latest
docker: Error response from daemon: Duplicate mount point: /mnt1.
See 'docker run --help'.

It looks like you can not have duplicate mount points (though you can have volume bindings which point to the same host directory). In this case a better model for the volume bindings would be to store them in something like a map where the key is the mount point. That, unfortunately, will require quite a few changes throughout the codebase (and I would advise against spending time on that at the moment).

While inefficient, the simplest solution would probably be to handle this logic within `<>` for now.

Reply:
OK so maybe best course is to...
- have the semigroup and monoid be "outlaws" for now
- open ticket to consider change to set/map like structure for the bindings?